loopy/transform/iname.py

```diff
@@ -305,7 +305,15 @@ def _split_iname_backend(kernel, iname_to_split,
     ins = _InameSplitter(rule_mapping_context, within,
             iname_to_split, outer_iname, inner_iname, new_loop_index)
 
-    kernel = ins.map_kernel(kernel)
+    from loopy.symbolic import get_dependencies
+    from loopy.kernel.instruction import Assignment
+
+    def check_insn_has_iname(kernel, insn, *args):
+        return not (isinstance(insn, Assignment) and
+                iname_to_split not in get_dependencies(insn.assignee) and
+                iname_to_split not in get_dependencies(insn.expression))
```

Comment:
You should probably also consider `within_inames`.

Reply:
`within_inames` is changed before this.